### DISCUSSION TITLE
Need Google Colab Tutorial for this

### DISCUSSION THREAD

**Discussion by ghost** [mindee/doctr#1088] [Q&A]
Hello, I hope you are doing well. I want a google colab tutorial for this repo

**Comment by felixdittrich92** [comment]
Hi @hammadyounas2008 :wave:,
we have some small Colab tutorials: https://mindee.github.io/doctr/notebooks.html

**Reply by ghost** [reply]
When i run this notebook on google colab 
get the error ModuleNotFoundError: No module named 'doctr'

**Reply by felixdittrich92** [reply]
You are right we need to update the tutorials :sweat_smile: 
1. Run Latest Stable Release Code Block 
2. Restart Runtime
3. Run Latest Stable Release Code Block again (first output should say python-doct[torch] 0.6.0)
4. Skip From Source
5. Run the other Code Blocks

**Reply by ghost** [reply]
I am using [[Export as PDF/A]](https://github.com/mindee/notebooks/blob/main/doctr/export_as_pdfa.ipynb) it gives the same error

**Reply by ghost** [reply]
Also getting this error
ModuleNotFoundError: No module named 'tf2onnx'

**Reply by felixdittrich92** [reply]
change first block to this: (as mentioned we need to update the tutorials)
```
#!pip3 install git+https://github.com/mindee/doctr.git
!pip3 install python-doctr[torch]
!pip3 install reportlab>=3.6.2
# optional if you want to merge multiple pdfs
!pip3 install PyPDF2==1.26.0
%env USE_TORCH=1
```
Run -> Restart Env -> Run again -> then you can run  the other blocks -> after running all in files you can download the PDF/A to check it locally

**Reply by ghost** [reply]
The saved pdf is not annotated? How can i download annotated pdf?

**Reply by felixdittrich92** [reply]
Mh PDF/A has an TextLayer so if you search words in an PDF-Reader it can find the words. What do you mean with annotated PDF ?

**Reply by ghost** [reply]
![download](https://user-images.githubusercontent.com/70446516/194280437-de790fe0-18e3-4c40-9bda-ac3106d8fff8.png)

Like this output can we save it in PDF format?

**Reply by felixdittrich92** [reply]
Ah ok no not directly but you can do it on your own with our output:

```
from tempfile import TemporaryDirectory
import os

from PIL import Image
import cv2

from doctr.io import DocumentFile
from doctr.models import ocr_predictor

doc = DocumentFile.from_pdf("/home/felix/Desktop/Versicherungsbedingungen-08-2021.pdf")
predictor = ocr_predictor(reco_arch="crnn_mobilenet_v3_small", det_arch="db_mobilenet_v3_large", pretrained=True).cuda()
res = predictor(doc)
export = res.export()

with TemporaryDirectory() as tmpdir:

    for page in export["pages"]:
        page_idx = page["page_idx"]
        height, width = page["dimensions"]
        for block in page["blocks"]:
            for line in block["lines"]:
                for word in line["words"]:
                    xmin = int(word['geometry'][0][0] * width)
                    ymin = int(word['geometry'][0][1] * height)
                    xmax = int(word['geometry'][1][0] * width)
                    ymax = int(word['geometry'][1][1] * height)
                    cv2.rectangle(doc[page_idx], (xmin, ymin), (xmax, ymax), (255, 0, 0), 2)

        cv2.imwrite(os.path.join(tmpdir, f'{page_idx}.png'), cv2.cvtColor(doc[page_idx], cv2.COLOR_BGR2RGB))

    saved_files = [os.path.join(tmpdir, file_path) for file_path in sorted(os.listdir(tmpdir))]
    images = [Image.open(file_path) for file_path in saved_files]
    images[0].save('overlayed.pdf', save_all=True, append_images=images[1:])
```

**Reply by felixdittrich92** [reply]
If you need any explanation let me know :+1:
